src/main/java/ir/sahab/kafkaconsumer/OffsetTracker.java

```diff
@@ -122,12 +122,21 @@ public PartitionTracker(long initialOffset) {
         boolean track(long offset) {
             long pageIndex = offsetToPage(offset);
             int margin = (int) (offset % pageSize);
-            if (pageTrackers.get(pageIndex) == null) {
+            PageTracker pageTracker = pageTrackers.get(pageIndex);
+            if (pageTracker == null) {
                 if (pageTrackers.size() >= maxOpenPagesPerPartition) {
                     return false;
                 }
-                pageTrackers.put(pageIndex, new PageTracker(pageSize, margin));
+                pageTracker = new PageTracker(pageSize, margin);
+                pageTrackers.put(pageIndex, pageTracker);
+                PageTracker prePage = pageTrackers.get(pageIndex - 1);
+                if (prePage != null) {
+                    if (prePage.markNoMoreTracks()) {
```

Comment:
```suggestion
                   prePage.markNoMoreTracks();
                    if (prePage.isCompleted()) {
```

Do not define return value for markNoMoreTracks() and instead define another method. (Just for the sake of more clarity)